mungegithub/mungers/publish_scripts/apimachinery_sync_from_kubernetes.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# overall flow
+# 1. fetch the current level of k8s.io/kubernetes
+# 2. check out the k8s.io/kubernetes HEAD into a separate branch
+# 3. rewrite the history on that branch to *only* include staging/src/k8s.io/apimachinery
+# 4. locate all commits between the last time we sync'ed and now
+# 5. switch back to the starting branch
+# 6. for each commit, cherry-pick it (which will keep authorship) into current branch
+# 7. update metadata files indicating which commits we've sync'ed to
+
+set -e
+
+if [ ! $# -eq 2 ]; then
+    echo "usage: apimachinery-sync-from-kubernetes.sh token netrc_dir. netrc_dir is expected to be absolute paths."
+    exit 1
+fi
+
+TOKEN="${1}"
+NETRCDIR="${2}"
+
+# set up github token
+echo "machine github.com login ${TOKEN}" > "${NETRCDIR}"/.netrc
+rm -f ~/.netrc
+ln -s "${NETRCDIR}"/.netrc ~/.netrc
+# set up github user
+git config --global user.email "k8s-publish-robot@users.noreply.github.com"
+git config --global user.name "Kubernetes Publisher"
```

Comment:
Is there any way to not do this? This is really user hostile, I can't manually run this script without it doing bad things to my git config. Can we split this script into bits that set up the bot's environment, and bits that actually do the publishing, so that a human can run the latter if they want to?